src/pages/SharedPage.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useCallback, useEffect, useState } from "react";
+import FolderInfo from "../components/FolderInfo";
+import SearchBar from "../components/SearchBar";
+import styles from "./SharedPage.module.css";
+import Card from "../components/Card";
+import { getFolderData } from "../api";
+import useAsync from "../hooks/useAsync";
+
+function SharedPage() {
+  const [folder, setFolder] = useState(null);
+  const [isApplying, ApplyingError, onApplyAsync] = useAsync(getFolderData);
+
+  const applyFolderInfoData = useCallback(async () => {
+    const folderData = await onApplyAsync();
+    if (!folderData) return;
+    setFolder(folderData);
+  }, [onApplyAsync]);
+
+  useEffect(() => {
+    applyFolderInfoData();
+  }, [applyFolderInfoData]);
+
+  if (ApplyingError) return <div>{ApplyingError.message}</div>;
+  return (
+    folder && (
+      <>
+        <FolderInfo folder={folder} />
+        <main>
+          <section className={styles.searchBarContainer}>
+            <SearchBar />
+          </section>
+          <section className={styles.cardContainer}>
+            {folder?.links.map((link) => (
```

Comment:
```suggestion
            {folder.links.map((link) => (
```
위에서 `folder &&`을 통해서 folder가 trusy임을 이미 체크 했기 때문에 optional chaining은 필요 없을것 같아요.